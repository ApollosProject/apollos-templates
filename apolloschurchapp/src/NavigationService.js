import { StackActions, NavigationActions } from 'react-navigation';

let _navigator;

const setTopLevelNavigator = (navigatorRef) => {
  _navigator = navigatorRef;
};

const navigate = (routeName, params) => {
  _navigator.dispatch(
    NavigationActions.navigate({
      routeName,
      params,
    })
  );
};

export const resetAction = ({ navigatorName, routeName }) =>
  StackActions.reset({
    index: 0,
    key: null,
    actions: [
      NavigationActions.navigate({
        routeName: navigatorName,
        action: NavigationActions.navigate({
          routeName,
        }),
      }),
    ],
  });

const resetToAuth = () => {
  _navigator.dispatch(
    resetAction({
      navigatorName: 'Auth',
      routeName: 'AuthSMSPhoneEntryConnected',
    })
  );
};

const goBack = (from) => {
  let key;
  if (from) {
    const route = _navigator.state.nav.routes.find((r) => r.routeName === from);
    if (route) ({ key } = route);
  }
  _navigator.dispatch(NavigationActions.back({ key }));
};

export default {
  setTopLevelNavigator,
  navigate,
  resetAction,
  goBack,
  resetToAuth,
};
