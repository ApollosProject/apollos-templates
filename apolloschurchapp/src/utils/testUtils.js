/* eslint-disable import/prefer-default-export */
import renderer from 'react-test-renderer';
import wait from 'waait';

export const renderWithApolloData = async (component) => {
  const tree = renderer.create(component);
  await wait(1);
  tree.update(component);
  return tree;
};
