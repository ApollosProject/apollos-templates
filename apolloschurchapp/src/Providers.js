import React from 'react';
import ApollosConfig from '@apollosproject/config';
import { Providers } from '@apollosproject/ui-kit';
import { AuthProvider } from '@apollosproject/ui-auth';
import { AnalyticsProvider } from '@apollosproject/ui-analytics';
import { MediaPlayerProvider } from '@apollosproject/ui-media-player';
import { NotificationsProvider } from '@apollosproject/ui-notifications';
import { LiveProvider } from '@apollosproject/ui-connected';

import NavigationService from './NavigationService';
import ClientProvider from './client';
import customTheme, { customIcons } from './theme';

const AppProviders = (props) => (
  <ClientProvider {...props}>
    <NotificationsProvider
      oneSignalKey={ApollosConfig.ONE_SIGNAL_KEY}
      navigate={NavigationService.navigate}
    >
      <AuthProvider
        navigateToAuth={() => NavigationService.navigate('Auth')}
        navigate={NavigationService.navigate}
        closeAuth={() => NavigationService.navigate('Onboarding')}
      >
        <MediaPlayerProvider>
          <AnalyticsProvider>
            <LiveProvider>
              <Providers
                themeInput={customTheme}
                iconInput={customIcons}
                {...props}
              />
            </LiveProvider>
          </AnalyticsProvider>
        </MediaPlayerProvider>
      </AuthProvider>
    </NotificationsProvider>
  </ClientProvider>
);

export default AppProviders;
