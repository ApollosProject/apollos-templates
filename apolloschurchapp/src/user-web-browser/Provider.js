import { createContext } from 'react';
import NavigationService from '../NavigationService';

// Can can either call the OpenUserWebView function directly, or call it using
// the Consumer. Your choice.
export const OpenUserWebView = ({ url }) =>
  NavigationService.navigate('UserWebBrowser', { url });

const { Provider, Consumer } = createContext(OpenUserWebView);

export { Provider as UserWebBrowserProvider };
export { Consumer as UserWebBrowserConsumer };
