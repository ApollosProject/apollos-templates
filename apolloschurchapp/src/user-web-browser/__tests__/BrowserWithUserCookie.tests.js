import React from 'react';
import { renderWithApolloData } from '../../utils/testUtils';
import Providers from '../../Providers';
import BrowserWithUserCookie, { WITH_USER_COOKIE } from '../index';
import { OpenUserWebView } from '../Provider';
import NavigationService from '../../NavigationService';

const mocks = [
  {
    request: { query: WITH_USER_COOKIE },
    result: {
      data: {
        __typename: 'AuthenticatedUser',
        currentUser: {
          __typename: 'Person',
          id: 'Person:123',
          rockToken: 'ABC',
        },
      },
    },
  },
];
const navigation = { navigate: jest.fn(), getParam: jest.fn() };

describe('the BrowserWithUserCookie component', () => {
  it('renders with a user', async () => {
    const tree = await renderWithApolloData(
      <Providers mocks={mocks}>
        <BrowserWithUserCookie navigation={navigation} />
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });
  it('renders with modal false', async () => {
    const tree = await renderWithApolloData(
      <Providers mocks={mocks}>
        <BrowserWithUserCookie modal={false} navigation={navigation} />
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });
});
describe('the OpenUserWebView', () => {
  jest.mock('NavigationService');
  NavigationService.navigate = jest.fn();
  it('navigtes', () => {
    OpenUserWebView({ url: 'fake.com' });
    expect(NavigationService.navigate).toBeCalledWith('UserWebBrowser', {
      url: 'fake.com',
    });
  });
});
