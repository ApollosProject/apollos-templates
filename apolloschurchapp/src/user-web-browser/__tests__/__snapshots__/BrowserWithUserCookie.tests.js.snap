// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the BrowserWithUserCookie component renders with a user 1`] = `
<BVLinearGradient
  colors={
    Array [
      4294967295,
      4294506484,
    ]
  }
  endPoint={
    Object {
      "x": 0.5,
      "y": 1,
    }
  }
  locations={null}
  startPoint={
    Object {
      "x": 0.5,
      "y": 0,
    }
  }
  style={
    Object {
      "borderRadius": 0,
      "flex": null,
      "height": "100%",
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "overflow": "hidden",
      }
    }
  >
    <RNCWKWebView
      cacheEnabled={true}
      messagingEnabled={false}
      onLoadingError={[Function]}
      onLoadingFinish={[Function]}
      onLoadingProgress={[Function]}
      onLoadingStart={[Function]}
      onMessage={[Function]}
      onShouldStartLoadWithRequest={[Function]}
      source={
        Object {
          "headers": Object {
            "Cookie": "ABC",
          },
          "uri": undefined,
        }
      }
      style={
        Array [
          Object {
            "flex": 1,
            "overflow": "hidden",
          },
          Object {
            "backgroundColor": "#ffffff",
          },
          undefined,
        ]
      }
      useSharedProcessPool={true}
    />
  </View>
  <View
    pointerEvents="box-none"
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingHorizontal": 16,
        "paddingVertical": 16,
        "position": "absolute",
        "top": 0,
        "width": "100%",
      }
    }
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight={32}
      bbWidth={32}
      fill="#303030"
      height={32}
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          undefined,
          null,
          Object {
            "flex": 0,
            "height": 32,
            "width": 32,
          },
        ]
      }
      vbHeight={24}
      vbWidth={24}
      width={32}
    >
      <RNSVGGroup
        fill={
          Array [
            0,
            4281348144,
          ]
        }
        fillOpacity={1}
        fillRule={1}
        font={Object {}}
        matrix={
          Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ]
        }
        opacity={1}
        originX={0}
        originY={0}
        propList={
          Array [
            "fill",
          ]
        }
        rotation={0}
        scaleX={1}
        scaleY={1}
        skewX={0}
        skewY={0}
        stroke={null}
        strokeDasharray={null}
        strokeDashoffset={null}
        strokeLinecap={0}
        strokeLinejoin={0}
        strokeMiterlimit={4}
        strokeOpacity={1}
        strokeWidth={1}
        vectorEffect={0}
        x={0}
        y={0}
      />
    </RNSVGSvgView>
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight={32}
      bbWidth={32}
      fill="#303030"
      height={32}
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          undefined,
          null,
          Object {
            "flex": 0,
            "height": 32,
            "width": 32,
          },
        ]
      }
      vbHeight={24}
      vbWidth={24}
      width={32}
    >
      <RNSVGGroup
        fill={
          Array [
            0,
            4281348144,
          ]
        }
        fillOpacity={1}
        fillRule={1}
        font={Object {}}
        matrix={
          Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ]
        }
        opacity={1}
        originX={0}
        originY={0}
        propList={
          Array [
            "fill",
          ]
        }
        rotation={0}
        scaleX={1}
        scaleY={1}
        skewX={0}
        skewY={0}
        stroke={null}
        strokeDasharray={null}
        strokeDashoffset={null}
        strokeLinecap={0}
        strokeLinejoin={0}
        strokeMiterlimit={4}
        strokeOpacity={1}
        strokeWidth={1}
        vectorEffect={0}
        x={0}
        y={0}
      />
    </RNSVGSvgView>
  </View>
</BVLinearGradient>
`;

exports[`the BrowserWithUserCookie component renders with modal false 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "overflow": "hidden",
    }
  }
>
  <RNCWKWebView
    cacheEnabled={true}
    messagingEnabled={false}
    onLoadingError={[Function]}
    onLoadingFinish={[Function]}
    onLoadingProgress={[Function]}
    onLoadingStart={[Function]}
    onMessage={[Function]}
    onShouldStartLoadWithRequest={[Function]}
    source={
      Object {
        "headers": Object {
          "Cookie": "ABC",
        },
        "uri": undefined,
      }
    }
    style={
      Array [
        Object {
          "flex": 1,
          "overflow": "hidden",
        },
        Object {
          "backgroundColor": "#ffffff",
        },
        undefined,
      ]
    }
    useSharedProcessPool={true}
  />
</View>
`;
